### ISSUE TITLE
HX-Swap in addition to HX-Retarget?

### ISSUE THREAD

**Issue by BoPeng** [bigskysoftware/htmx#933]
There are cases when I would like to cancel htmx swapping. For example,  in #932 the server swaps the form with a form with error messages (`hx-target="my-form"`), but sends a signal when the form is successfully processed (showing a confirmation modal form without swapping).

I was using

```
HX-Retarget="none"
``` 
without giving it a second thought and the code worked, until I saw error messages in the console saying invalid target. It turns out that

https://github.com/bigskysoftware/htmx/blob/2872b7dbc4d7e996204268571b0f34c8dfcf2799/src/htmx.js#L2810

tries to find an element `none` and failed.

Since what I actually need to change is to set `hx-swap="none"`, can there be a header called `HX-Swap`? In addition to setting `HX-Swap` to "none" to stop swapping, it is possible to do other things such as 

When using a form to submit a post in a forum,
1. If form fails, replace the form with a form with error message (`hx-swap="outerHTML" hx-target="my-form")
2. If form passes, set `HX-Retarget="#post-list"`, `HX-Swap="beforeend"` to append the post to the list of posts.

The bottom line is that even just for completeness, there could be an `HX-Swap` changing `hx-swap` in addition to `HX-Retarget` changing `hx-target`.

**Comment by applecat** [comment]
I am constantly faced with the need of changing `hx-swap` parameter.

For instance, I have a simple voting widget [-1 result +1], which works only for authenticated users. So in most cases I need `hx-swap="innerHTML"`, but when unauthorized user clicks button I'd like to show him a login modal dialog and than I need change `hx-target` to `"body"` and `hx-swap` to `"beforeend"`. There is great `HX-Retarget` header, but there isn't `HX-Reswap`.

**Comment by BoPeng** [comment]
@1cg I encountered another scenario today when `HX-Swap` is needed. Basically, the orignal `hx-swap` uses the default `innerHTML` but I cannot do the same to the target set by `HX-Retarget`. I could try to wrap the new target with a wrapping `div` and use `innerHTML` as well, but a new header `HX-Swap` or `HX-Reswap` is a more natural approach.

I had a brief look at the source code and there appears to be an existing mechanism to override `hx-swap`

https://github.com/bigskysoftware/htmx/blob/d997834842cc688e6abcbd29999a57190baefb0a/src/htmx.js#L2852-L2853

in the same function after `hx-target` is overriden

https://github.com/bigskysoftware/htmx/blob/d997834842cc688e6abcbd29999a57190baefb0a/src/htmx.js#L2809-L2810

So maybe this can be an easy feature to add?

**Comment by BoPeng** [comment]
THANKS!

**Comment by ulyssejdv** [comment]
Hi ! Is this feature exists in version 1.8.0 ?

I try something like this :

```python
http_response.content = "error"
http_response["HX-Retarget"] = "#error-area"
http_response["HX-Reswap"] = "innerHTML"
```
in my Django view but it has no effect.

With 

```html
<div id="error-area">
</div>
```

in the template

**Comment by 2x2xplz** [comment]
Thank you @1cg and @mayowa , HX-Retarget and HX-Reswap add enormous flexibility to this already-fantastic library. Great job!

**Comment by residentcode** [comment]
> Hi ! Is this feature exists in version 1.8.0 ?
> 
> I try something like this :
> 
> ```python
> http_response.content = "error"
> http_response["HX-Retarget"] = "#error-area"
> http_response["HX-Reswap"] = "innerHTML"
> ```
> 
> in my Django view but it has no effect.
> 
> With
> 
> ```
> <div id="error-area">
> </div>
> ```
> 
> in the template

`response = HttpResponse('<small remove-me="5s" class="text-pink-600">text</small>')
response['HX-Retarget'] = '#alert_text'
return response`

confirmed works
